### DISCUSSION TITLE
How to Send and Receive messages using CEF profile

### DISCUSSION THREAD

**Discussion by duke-nuke** [phax/phase4#261] [Q&A]
Hi 

I am already sending / receiving einvoices using Peppol profile , now we need to send einvoices to europe but that should be using CEF profile, 
Can we receive CEF einvoice using same instance of as4 which is running with peppol profile or We need to start a new AS4 instance with CEF profile ?  further same validation rules are applied on CEF documents and Peppol documents ? 

is there any example of CEF AS4 server and client ? the current client can handle delivery of CEF einvoices to europe clients ? 

regards

**Comment by phax** [comment]
Hello @duke-nuke,
Well, handling a different AS4 profile using the the same endpoint is possible, but is not totally straight forward.
Usually there are parameters inside a message that lets you choose the AS4 profile, but I recommend to use an "outside" determinator like a separate hostname.
So e.g. running the Peppol AS4 profile at "host/peppol/as4" and the CEF profile at "host/cef/as4" but both served with the same AS4 servlet.
I will setup a separate Wiki page where I will describe the best practices to do so. Currently there is no simple solution available, but it is of course possible.
For the sender side it is straight forward, but for the receiver side it is a bit more tricky.... I will post the link to the Wiki page here, once it is available.

**Reply by duke-nuke** [reply]
Hi thanks for your reply 
European commission has given us different files to setup  as4 
Pmode.xml and couple of certs do we need to setup the pmode.xml somewhere ? This  file is in domibus schema . 

Another question is how can we send einvoice to cef using your library is there any guide
Thanks

**Reply by phax** [reply]
Yes, this `Pmode.xml` is Domibus specific if I am not mistaken. The certificates you need to install in a respective keystore and trust store.
An no - regarding the specific eInvoices to the EC: I have no further information at hand. Can you eventually share a link to the specifics?

I started the page to multi-profile handling at https://github.com/phax/phase4/wiki/Multi-Profile-Handling but the content is not complete yet :)

**Comment by duke-nuke** [comment]
Hi thanks for your reply 

We have to certify our einvoicing solution is able to send and receive einvoices to eu cef
You can find more details here

 https://ec.europa.eu/digital-building-blocks/sites/display/DIGITAL/eInvoicing

Thanks again

**Reply by phax** [reply]
I am aware of this page, but this is a very "generic" page on eInvoicing in Europe. It relates to the EN 16931 standard etc. but is not receiver specific.
If you need to send to the EC itself, there needs to be a more detailed page that outlines the requirement for the specific receiver "EC".
Hope my questions make sense....
